pkg/snowflake/function.go

```diff
@@ -140,6 +161,13 @@ func Function(db, schema, name string, argTypes []string) *FunctionBuilder {
 	}
 }
 
+func (pb *FunctionBuilder) UseWarehouse() (string, error) {
+	var q strings.Builder
+
+	q.WriteString(fmt.Sprintf("USE WAREHOUSE %v", pb.warehouse))
```

Comment:
this shouldn't be set on a single resource but on the session. within tf, it should be set per instance of the provider.

Reply:
cc @sfc-gh-swinkler to clarify where the warehouse goes in the other piece

Reply:
@sfc-gh-afedorov it is not set on the instance of the provider, since the provider can create multiple warehouses and not one of them can be considered a default warehouse.

i don't like having to put this here, but in discussion with @sfc-gh-kmaurya this is the best place to put it.